### PR TITLE
Best Practices PMD rules

### PR DESCRIPTION
**Pull request by alovew**
Implement the following Best Practices PMD rules:

LiteralsFirstInComparisons
LooseCoupling
PreserveStackTrace
SwitchStmtsShouldHaveDefault
SystemPrintln
UseCollectionIsEmpty
UseVarargs

### CODE REVIEW

File: airbyte-metrics/metrics-lib/src/main/java/io/airbyte/metrics/lib/MetricClientFactory.java [hunk 1]
```diff
@@ -51,14 +51,14 @@ public synchronized static MetricClient getMetricClient() {
...
    * @param metricEmittingApp the name of the app which the metric will be running under.
    */
-  public static synchronized void initialize(MetricEmittingApp metricEmittingApp) {
+  public static synchronized void initialize(final MetricEmittingApp metricEmittingApp) {
     if (metricClient != null) {
       throw new RuntimeException("You cannot initialize configuration more than once.");
     }
 
-    if (configs.getMetricClient().equals(DATADOG_METRIC_CLIENT)) {
+    if (DATADOG_METRIC_CLIENT.equals(configs.getMetricClient())) {
```
Comment (alovew) [inline airbyte-metrics/metrics-lib/src/main/java/io/airbyte/metrics/lib/MetricClientFactory.java:59]: actually saw a bug in dev recently due to this, so this is good to fix.

File: airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/JobNotifier.java [hunk 2]
```diff
@@ -101,28 +101,29 @@ private void notifyJob(final String reason,
...
-                LOGGER.warn("Failed to successfully notify auto-disable connection: {}", notification);
-              }
-              break;
-            case CONNECTION_DISABLED_WARNING_NOTIFICATION:
-              if (!notificationClient.notifyConnectionDisableWarning(workspace.getEmail(), sourceConnector, destinationConnector, jobDescription,
-                  workspaceId, connectionId)) {
-                LOGGER.warn("Failed to successfully notify auto-disable connection warning: {}", notification);
-              }
+
+          if (FAILURE_NOTIFICATION == action) {
```
Comment (alovew) [inline airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/JobNotifier.java:105]: looking at this again, thinking maybe I should use a default and log something like "unknown action"?

File: airbyte-test-utils/src/main/java/io/airbyte/test/airbyte_test_container/AirbyteTestContainer.java [hunk 3]
```diff
@@ -155,10 +155,11 @@ private Consumer<OutputFrame> logConsumer(final String service, final Consumer<S
         final String message = prependService(service, log.replace("\n", ""));
         switch (c.getType()) {
           // prefer matching log levels from docker containers with log levels in logger.
-          case STDOUT -> LOGGER.info(message);
           case STDERR -> LOGGER.error(message);
           // assumption that this is an empty frame that connotes the container exiting.
           case END -> LOGGER.error(service + " stopped!!!");
+          // default includes STDOUT and anything else
+          default -> LOGGER.info(message);
```
Comment (alovew) [inline airbyte-test-utils/src/main/java/io/airbyte/test/airbyte_test_container/AirbyteTestContainer.java:162]: opinions on setting this as the default?

File: airbyte-workers/src/main/java/io/airbyte/workers/internal/DefaultAirbyteStreamFactory.java [hunk 4]
```diff
@@ -94,9 +94,9 @@ private void internalLog(final AirbyteLogMessage logMessage) {
     switch (logMessage.getLevel()) {
       case FATAL, ERROR -> logger.error(logMessage.getMessage());
       case WARN -> logger.warn(logMessage.getMessage());
-      case INFO -> logger.info(logMessage.getMessage());
       case DEBUG -> logger.debug(logMessage.getMessage());
       case TRACE -> logger.trace(logMessage.getMessage());
+      default -> logger.info(logMessage.getMessage());
```
Comment (alovew) [inline airbyte-workers/src/main/java/io/airbyte/workers/internal/DefaultAirbyteStreamFactory.java:99]: opinions on this default? (similar to above)

File: airbyte-test-utils/src/main/java/io/airbyte/test/airbyte_test_container/AirbyteTestContainer.java (hunk 3 above)
Comment (jdpgrailsdev) [reply airbyte-test-utils/src/main/java/io/airbyte/test/airbyte_test_container/AirbyteTestContainer.java:162]: That seems fine.  Worst case scenario is that we get a few extra cases logged to INFO.

File: airbyte-workers/src/main/java/io/airbyte/workers/internal/DefaultAirbyteStreamFactory.java (hunk 4 above)
Comment (jdpgrailsdev) [reply airbyte-workers/src/main/java/io/airbyte/workers/internal/DefaultAirbyteStreamFactory.java:99]: This seems fine.

**Review by jdpgrailsdev — APPROVED** [review]
:shipit:

**Pull request merged**